src/math/p_ceil.c

```diff
@@ -0,0 +1,29 @@
+#include <pal.h>
+/**
+*
+* Returns round up of input vector a.
+*
+*
+* @param a     Pointer to input vector
+*
+* @param c     Pointer to output vector
+*
+* @param n     Size of 'a' and 'c' vector.
+*
+* @return      None
+*
+*/
+
+void p_ceil_f32(const float *a, float *c, int n)
+{
+    int i; float x;
+    for ( i = 0; i < n; i++)
+      {
+       const float *pa = (a+i);
+             float *pc = (c+i);
+       x = *pa;
+       int j= (int) x;
+       if (j < x) j++;
```

Comment:
`(int)x + ((int)x < x)` would be simpler and avoids branching ([ref](https://en.wikipedia.org/wiki/Branch_%28computer_science%29#Performance_problems_with_branch_instructions))

also, I think both `p_ceil()` and `p_floor()` could be macros instead of functions.